ingestion_program/ingestion_parallel.py

```diff
@@ -286,10 +286,10 @@ def show_duration(self):
         print(f"[✔] Total duration: {self.get_duration()}")
         print("---------------------------------")
 
-    def save_duration(self):
+    def save_duration(self, output_dir=None):
```

Comment:
This seems to be duplicated between the parallel and sequential version. Common base class?

Reply:
Yeah ingestion parallel has many things similar to ingestion serial. But I am not personally a fan of inheritance. But if that makes cleaner sure. What do you think @ihsaan-ullah ?

Reply:
Lets keep them sperate for now. Changing something may break something else. In the next update we can clean and remove repetition